runtime/lua/vim/filetype/detect.lua

```diff
@@ -236,29 +580,135 @@ function M.mms(bufnr)
   vim.bo[bufnr].filetype = 'mmix'
 end
 
-function M.mod(path, bufnr) end
+-- Returns true if file content looks like LambdaProlog
+local function is_lprolog(bufnr)
+  -- Skip apparent comments and blank lines, what looks like
+  -- LambdaProlog comment may be RAPID header
+  for _, line in ipairs(getlines(bufnr, 1, -1)) do
+    -- The second pattern matches a LambdaProlog comment
+    if not findany(line, { '^%s*$', '^%s*%%' }) then
+      -- The pattern must not catch a go.mod file
+      return findany(line, { 'module%s+%w+%s*.%s*%%', 'module%s+%w+%s*.%s*$' })
+    end
+  end
+end
+
+-- Determine if *.mod is ABB RAPID, LambdaProlog, Modula-2, Modsim III or go.mod
+function M.mod(path, bufnr)
+  if vim.g.filetype_mod then
+    vim.bo[bufnr].filetype = vim.g.filetype_mod
+  elseif is_lprolog(bufnr) then
+    vim.bo[bufnr].filetype = 'lprolog'
+  elseif findany(nextnonblank(bufnr, 1):lower(), { 'module%s+%w+%s*;', '^%s*%(%*' }) then
```

Comment:
`MODULE` is uppercase in filetype.vim so let's assume it's case sensitive here. And since it uses `\<` let's just copy the pattern directly:

```suggestion
  elseif vim.regex([[\%(\<MODULE\s\+\w\+\s*;\|^\s*(\*\)]]):match_str(nextnonblank(bufnr, 1)) then
```